app/uk/gov/hmrc/hec/services/IFService.scala

```diff
@@ -119,40 +118,78 @@ class IFServiceImpl @Inject() (
           httpResponse
             .parseJSON[RawCTSuccessResponse]
             .leftMap(e => BackendError(Error(e)))
-            .flatMap(response =>
-              Either
-                .fromOption(
-                  CTStatus.fromString(response.returnStatus),
-                  BackendError(Error(s"Could not parse success return status ${response.returnStatus}"))
-                )
-                .map(CTStatusResponse(utr, startDate, endDate, _, response.accountingPeriods))
-            )
+            .flatMap(ctAccountingPeriodsValidation(_).map { accountingPeriods =>
+              CTStatusResponse(utr, startDate, endDate, latestAccountingPeriod(accountingPeriods))
+            })
         } else {
           handleErrorPath(httpResponse, utrType = "CT")
         }
       }
+
+  private def latestAccountingPeriod(accountingPeriods: List[AccountingPeriod]): Option[AccountingPeriod] =
+    accountingPeriods.sortWith { case (a1, a2) => a1.endDate.isAfter(a2.endDate) }.headOption
+
+  private def ctAccountingPeriodsValidation(
+    response: RawCTSuccessResponse
+  ): Either[BackendError, List[AccountingPeriod]]                                                         =
+    if (response.returnStatus === "0") // 0 = successful
```

Comment:
would it be worth using a sealed trait for the return status even if it means a bit of repetition? so we wouldn't need to qualify the value with a comment

Reply:
I thought about that, then I thought it'd be extra code to maintain for these small number of lines of code. But it might be worth it in the long run just for clarity's sake - I'll add it in then :+1:

Reply:
updated just now